src/js/media-controller.js

```diff
@@ -880,6 +936,28 @@ const monitorForMediaStateReceivers = (
   return unsubscribe;
 };
 
+let testMediaEl;
+export const getTestMediaEl = (nodeName = "video") => {
+  if (testMediaEl) return testMediaEl;
+  if (typeof window !== "undefined") {
+    testMediaEl = document.createElement(nodeName);
+  }
+  return testMediaEl;
+};
+
+export const hasVolumeSupportAsync = async (
```

Comment:
We should execute this test immediately and store the results instead of waiting for a media-controller init. There's essentially two stages to the volume test:
1. Does the underlying platform support it (i.e. even `<youtube-video>` will be limited by the results)
2. Does the current media support volume. Even if the platform does, the specific media may not for some reason. We could choose to kick this one down the road, but just calling it out as a possibility.

Reply:
I'll have to test that in an SSR environment, but I get why you're calling that out. Also, all of the async media elements will add complexities here, so I'm glad you're calling that out.

Reply:
>  Promise.then()
I think I did in a previous incarnation? But def worth trying again, since I can't recall the details (also I prefer promise.resolve or requestAnimationFrame over setTimeout 0)